pkg/pullrequest/api.go

```diff
@@ -272,15 +272,22 @@ func (h *Handler) uploadStatuses(ctx context.Context, statuses []*scm.Status, sh
 	}
 
 	// Index the statuses so we can avoid sending them if they already exist.
-	csMap := map[string]scm.State{}
+	csMap := map[string]scm.Status{}
 	for _, s := range cs {
-		csMap[s.Label] = s.State
+		// Statuses are returned by the API in reverse chronological order:
+		// https://developer.github.com/v3/repos/statuses/#list-statuses-for-a-specific-ref
```

Comment:
We shouldn't tie semantics directly to the GitHub API (e.g. does this also hold true for GitLab?)

We may need another mechanism to determine "most recent".

Reply:
Good point. I will check what gitlab does here

Reply:
The problem here is that the go-scm API does not expose any other mean of selecting the best version of a status:
https://github.com/jenkins-x/go-scm/blob/838e28f71c1eb2c8b4486e2b50bc498c8a5cfd0f/scm/repo.go#L91-L96

I will file an issue on go-scm side, I think the best solution would be for them to do the filtering in the GitHub driver. I would like to keep this code until the patches merges on go-scm side, with an updated comment.

Reply:
https://github.com/jenkins-x/go-scm/issues/79

Reply:
and https://github.com/jenkins-x/go-scm/pull/80

Reply:
Done